### ISSUE TITLE
why mask is all zero?

### ISSUE THREAD

**Issue by mmjwxbc** [jingyaogong/minimind#308]
<img width="659" alt="Image" src="https://github.com/user-attachments/assets/4cab82ae-5352-4343-b593-1a18a66c75fb" />

```python
 ref_probs = logits_to_probs(ref_logits, y)
ref_probs = ref_probs * mask
```

I think in the above codeblock, ref_probs = ref_prbos * [1,1,1,0,0] should like this. But the mask is [0,0,0,0,0,0], I think it is strange!!!
